shared/chat/conversation/info-panel/container.tsx

```diff
@@ -329,93 +337,59 @@ type SelectorOwnProps = Container.RouteProps<{
   attachmentview: RPCChatTypes.GalleryItemTyp
 }>
 
-const mapStateToSelectorProps = (state: Container.TypedState, ownProps: SelectorOwnProps) => {
-  const conversationIDKey: Types.ConversationIDKey = Container.getRouteProps(
-    ownProps,
-    'conversationIDKey',
-    Constants.noConversationIDKey
-  )
-  const meta = Constants.getMeta(state, conversationIDKey)
-  const selectedTab = Container.getRouteProps(ownProps, 'tab', null)
-  const selectedAttachmentView = Container.getRouteProps(
-    ownProps,
-    'attachmentview',
-    RPCChatTypes.GalleryItemTyp.media
-  )
-  return {
-    conversationIDKey,
-    selectedAttachmentView,
-    selectedTab,
-    shouldNavigateOut: meta.conversationIDKey === Constants.noConversationIDKey,
-  }
-}
-
-const mapDispatchToSelectorProps = (dispatch, {navigation}) => ({
-  // Used by HeaderHoc.
-  onBack: () => dispatch(Chat2Gen.createToggleInfoPanel()),
-  onGoToInbox: () => dispatch(Chat2Gen.createNavigateToInbox()),
-  onSelectAttachmentView: view => navigation.setParams({attachmentview: view}),
-  onSelectTab: (tab: Panel) => navigation.setParams({tab}),
-})
-
-const mergeSelectorProps = (stateProps, dispatchProps, _: SelectorOwnProps) => ({
-  conversationIDKey: stateProps.conversationIDKey,
-  onBack: dispatchProps.onBack,
-  onGoToInbox: dispatchProps.onGoToInbox,
-  onSelectAttachmentView: dispatchProps.onSelectAttachmentView,
-  onSelectTab: dispatchProps.onSelectTab,
-  selectedAttachmentView: stateProps.selectedAttachmentView,
-  selectedTab: stateProps.selectedTab,
-  shouldNavigateOut: stateProps.shouldNavigateOut,
-})
-
 type Props = {
   conversationIDKey: Types.ConversationIDKey
   onBack: () => void
   onGoToInbox: () => void
-  onSelectTab: (t: Panel) => void
-  selectedTab: Panel | null
-  onSelectAttachmentView: (typ: RPCChatTypes.GalleryItemTyp) => void
-  selectedAttachmentView: RPCChatTypes.GalleryItemTyp
   shouldNavigateOut: boolean
 }
 
-class InfoPanelSelector extends React.PureComponent<Props> {
-  componentDidUpdate(prevProps) {
-    if (!prevProps.shouldNavigateOut && this.props.shouldNavigateOut) {
-      this.props.onGoToInbox()
-    }
-  }
-  render() {
-    if (!this.props.conversationIDKey) {
-      return null
-    }
+const InfoPanelSelector = (props: Props) => {
+  const prevShouldNavigateOut = Container.usePrevious(props.shouldNavigateOut)
+  const {shouldNavigateOut, onGoToInbox} = props
+  React.useEffect(() => {
+    !prevShouldNavigateOut && shouldNavigateOut && onGoToInbox()
+  }, [prevShouldNavigateOut, shouldNavigateOut, onGoToInbox])
+  const [selectedTab, onSelectTab] = React.useState<Panel | null>(null)
```

Comment:
I think we initialize which tab to open through routeprops. e.g. clicking "manage retention" in chat takes you to the settings tab. I don't see us using this for `selectedAttachmentView` though.